app/code/Magento/Sales/Model/Order/Invoice.php

```diff
@@ -793,6 +794,13 @@ public function getItems()
             }
             $this->setData(InvoiceInterface::ITEMS, $collection->getItems());
         }
+        if ($invoiceItemCheck) {
+            $items = $this->getData(InvoiceInterface::ITEMS);
+            if ($items instanceof \Magento\Sales\Model\ResourceModel\Order\Invoice\Item\Collection) {
```

Comment:
Have you tested that? O.o
```suggestion
            if ($items instanceof \Magento\Sales\Model\ResourceModel\Order\Invoice\Item\Collection::class) {
```

Reply:
but that's correct, you can check any other condition.
Here is the example:
<img width="858" alt="m2git  ~:Public:projects:m2git  -  :app:code:Magento:Catalog:Model:Design php 2020-03-20 20-20-08" src="https://user-images.githubusercontent.com/10618641/77198944-41602b80-6ae8-11ea-8e0a-a5f77745f589.png">